### ISSUE TITLE
[BUG] Renovate bot suggests update to version 1.8.1259 from one year ago, latest currently is 1.2.2

### ISSUE THREAD

**Issue by floryn08** [linuxserver/docker-prowlarr#42]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Renovate bot suggests updating to version 1.8.1259 from one year ago, the latest currently is 1.2.2. Blocking this merge request by renovate prevents updating further.

### Expected Behavior

The 1.8.1259 tag should be removed from the container registry to avoid being picked up as a newer version.

### Steps To Reproduce

1. have a docker-compose file with prowlarr
```
 prowlarr:
    image: ghcr.io/linuxserver/prowlarr:1.2.2
    container_name: prowlarr
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=${TIMEZONE}
    volumes:
      - ${VOLUMES_PATH}/media-management/prowlarr:/config
    ports:
      - 9696:9696
    restart: unless-stopped
    networks:
      - media-management-network
  ```
2. configure renovate bot for repository
3. renovate bot creates merge request for update to 1.8.1259 which is an incorrect tag from one year ago

### Environment

```markdown
n/a
```

### CPU architecture

arm64

### Docker creation

```bash
n/a
```

### Container logs

```bash
n/a
```

Link to tag: https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1259/images/sha256-9af25e5a3d9ffae44726ecf4d1e2db21b3c389ff8c2d242e295988b7f3d14a49?context=explore

**Comment by Roxedus** [comment]
We don't have a history for deleting old tags, and we do not intend to start with this one-off.

**Comment by floryn08** [comment]
I can understand that. But in this case that release was incorrectly tagged as a major version when in fact it should have been 0.1.8.1259.

![image](https://user-images.githubusercontent.com/11478717/224995107-ddc213c2-aafe-4105-a1fb-79e744067306.png)

**Comment by Roxedus** [comment]
Which comes back to us not officially supporting, promoting or announcing semver tagging. This tagging scheme is just something we did when the upstream tag had three digits.

**Comment by PortableProgrammer** [comment]
> We don't have a history for deleting old tags, and we do not intend to start with this one-off. 

> Which comes back to us not officially supporting, promoting or announcing semver tagging. This tagging scheme is just something we did when the upstream tag had three digits.

These... are incredibly shortsighted responses. It's all fine and dandy to say, but what are you going to do when you hit `1.8.x` legitimately? Will you _then_ have to perform a one-off delete? Why not implement an actual deprecation policy, or even an actual semver policy? I mean, you've already mentioned that it was an artifact of the upstream's invalid semver, so it wouldn't necessarily even need to be breaking your own unwritten rules, it would just be a nice thing to do for the community to clean up obviously wrong tags. Your `develop` and `nightly` tags got it right, but your stable version tagging failed.

For those looking to work around this, I was able to configure Flux to filter tags down to `^\d*\.?\d*\.?\d{0,2}$` before applying semver, which is still not great; it presumes this repo never legitimately hits a triple-digit rev/patch before the next minor or major bump, and that we'll never have duplicated tags.

@floryn08 If you're still having issues here, I haven't used Renovate Bot, but a quick glance at https://docs.renovatebot.com/modules/versioning/ makes me think you could utilize the regex pattern to achieve the same result, with the same caveats.

**Comment by Roxedus** [comment]
Docker unfortunately overwrite old tags with asking, so we will not have any issues there.

Implementing semver just isnt feasible across all our image, and we do not want to maintain logic for one-offs.

**Comment by aptalca** [comment]
> what are you going to do when you hit 1.8.x legitimately?

Nothing. Why would we do anything?

We never supported semver, never claimed to have supported it. We never supported renovate or similar tools that solely rely on semver, never claimed to have supported them.

We did experiment with pushing pseudo-semver tags. The experiment wasn't a success. We never publicized it. We never promoted it.

Semver is simply not appropriate for our images because while it is appropriate for an upstream project that only has to version their own specific project, what we ship is a combination of things that inlcude the upstream project, an entire OS, our various init and service scripts, and lots of dependency packages we install in there. Sure we can tag it with the upstream project's semver tag to make you happy, but guess what, when we rebase the OS and/or update the various dependency packages, or when we significantly restructure our init scripts and/or defaults we ship, there may very well be major breaking changes. You as an end user will see the same exact major-minor from the upstream and have a false sense of security when in reality your stuff breaks completely.

That's why we don't support or claim to support semver or renovate.

The tags we support and promote are in the following format:
`<branch name if not master or main>-<upstream version>-<our build tag ie. lsXX>`
we also promote the following if one wants to pin to a specific upstream version (no further os package updates once the upstream moves onto a new version):
`version-<branch name if not master or main>-<upstream version>`

**Comment by PortableProgrammer** [comment]
Thank you for your reply.

> The tags we support and promote are in the following format:
> `<branch name if not master or main>-<upstream version>-<our build tag ie. lsXX>`

That's fair, and if it were consistent, it would be perfect. I'll retract my statement about implementing your own semver policy; you're absolutely right, it's not your responsibility, it's not really feasible, and it would definitely introduce bigger issues if you were to try. I think an image deletion policy would still be a good idea, since that may have negated this issue, but that's also outside the bounds of this conversation.

**Semver aside**, the problem here is that the [build](https://hub.docker.com/r/linuxserver/prowlarr/tags?page=1&name=1.8.1259) referenced by this bug report _is not in compliance with your own policy_. The upstream version on the build is `0.1.8.1259`, so the stable multiarch tags, per your comment and the only [published standard](https://www.linuxserver.io/blog/2019-02-21-the-lsio-pipeline-project#the-great-latest-debate-now-you-have-a-choice) I could find, should be `0.1.8.1259`, `0.1.8.1259-lsXX`, and `version-0.1.8.1259`, any of which would have negated this entire bug report, whether or not we were attempting to use semver or Renovate/Flux or any other automated solution.

Ultimately, what we're trying to say is that you have a standard, but it wasn't applied consistently. This caused confusion that could have been easily remedied when it was detected (and after you admitted that it was indeed incorrect), but to simply shrug and say "we don't support or claim to support semver" is to ignore the entire point.

We have identified several ways to work around this, and I don't want to beat this particular horse further into dust than I already have, so I'll leave it there.

**Comment by aptalca** [comment]
> not in compliance with your own policy

I don't know which build you're referring to.
If you're referring to the lone image tagged [1.8.1259](https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1259/images/sha256-9af25e5a3d9ffae44726ecf4d1e2db21b3c389ff8c2d242e295988b7f3d14a49?context=explore) from a year ago, then I'd like to remind you once again that that `tag` is not one of the tags I mentioned as supported or promoted in my above post. It is an artifact of our unsuccessful pseudo-semver experiment. Thus, outside of the published standard you referred to.

For version `0.1.8.1259`, we do have [nightly-0.1.8.1259-ls231](https://hub.docker.com/layers/linuxserver/prowlarr/nightly-0.1.8.1259-ls231/images/sha256-4bee5e7626de33e4ab456c8e432b30528a003a024577264c2fa1c2115ac3dbde?context=explore) and [nightly-version-0.1.8.1259](https://hub.docker.com/layers/linuxserver/prowlarr/nightly-version-0.1.8.1259/images/sha256-ed8b0bd0e1276a3a63fff9f68a574b466960f76cc26ee5b9e550e4ab746739bf?context=explore) which fit the list of tags we support and promote.

If and when there is a stable build with version `0.1.8.125`, we will push images tagged `0.1.8.125-lsXX` and `version-0.1.8.125`

**Comment by PortableProgrammer** [comment]
Apologies, I mistakenly used `build` instead of `tag`. You're right, I was referring to the image tagged [`1.8.1259`](https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1259/images/sha256-9af25e5a3d9ffae44726ecf4d1e2db21b3c389ff8c2d242e295988b7f3d14a49?context=explore).

Furthermore, while the `nightly` tags do indeed conform to the standard, I was explicitly referring to `stable` tags, which are what we would expect an end user to pull. Absent any other indications, and following the standard, an end user could conceivably expect that `1.8.1259`, when viewed through the lens of upstream's versioning (flawed as it is), would be a later version than [`1.3.2`](https://hub.docker.com/layers/linuxserver/prowlarr/1.3.2/images/sha256-97031616ca49633a517e93b9d7fadaa74ee83bf3b0f7ebd36217243341a2aa8c?context=explore) / [`version-1.3.2.3006`](https://hub.docker.com/layers/linuxserver/prowlarr/version-1.3.2.3006/images/sha256-97031616ca49633a517e93b9d7fadaa74ee83bf3b0f7ebd36217243341a2aa8c?context=explore) / [`1.3.2.3006-ls12`](https://hub.docker.com/layers/linuxserver/prowlarr/1.3.2.3006-ls12/images/sha256-97031616ca49633a517e93b9d7fadaa74ee83bf3b0f7ebd36217243341a2aa8c?context=explore), which are the current `stable` tags.

I'm not entirely certain how to make my point any clearer: you have a standard, you tried something new, it was unsuccessful, and you're left with some hanging tags that don't conform to the standard. They should have been removed at the time the experiment was deemed unsuccessful. They can probably _still_ be removed, 16 months later, without serious breakage, given that they appear to have been based on [pre-release versions](https://github.com/Prowlarr/Prowlarr/releases/tag/v0.1.8.1231).

Ultimately, this still doesn't have anything to do with semver explicitly; it has everything to do with the fact that the experimental tagging process did not faithfully replicate upstream's versioning, leading to a tag that significantly diverges from upstream's versioning, causing end user confusion.

I found the following non-conforming tags, which will cause confusion until upstream achieves `1.8.1259.1` or `1.9.0.0`, based on their current versioning standard.
* [`1.8.1259`](https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1259/images/sha256-9af25e5a3d9ffae44726ecf4d1e2db21b3c389ff8c2d242e295988b7f3d14a49?context=explore)
* [`1.8.1253`](https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1253/images/sha256-7d2ac47fd0f20139bf4c7796d3e382aeb114d7f4998f14a8d24d6779ed2e8fee?context=explore)
* [`1.8.1238`](https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1238/images/sha256-2001d340781908c3485f3decded73ac55865469e4f27daa909d2e0901c9d6948?context=explore)
* [`1.8.1234`](https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1234/images/sha256-9b4e9cd1e27395f55b405f91921054264b42d37cc338941ff8fe33dd214bed21?context=explore)
* [`1.8.1232`](https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1232/images/sha256-074dbe7159ffbeede7117441b03cf5b94e4160505f2423cbad948cc4b36d391f?context=explore)
* [`1.8.1231`](https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1231/images/sha256-05cc0808adb3c987ab4dc9cd38de46614d59a4c1209f6c8f6b37bbcbf1e1b136?context=explore)
* [`1.8.1229`](https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1229/images/sha256-91b5507a1463a6625c7984790eab9f4728d0d55b398b93ee7711ae7433920515?context=explore)
* [`1.8.1223`](https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1223/images/sha256-c8adea3509ba8d29f36676c0f95737b64c43b136253c3ae8fd46f42f9ffa8dcd?context=explore)
* [`1.8.1221`](https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1221/images/sha256-53211778082ed68a54a117195910fc8c718263ac72ccee6794cfd5cca80b2bdc?context=explore)
* [`1.8.1217`](https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1217/images/sha256-678cda9ad19760df48f2d70f12759673796e7d47b0bf6e89b69a50f0257215d0?context=explore)
* [`1.8.1216`](https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1216/images/sha256-0fe7b275bb131d6ed33bdf4cc49df34b113287fef01a5b661e549d1a753d9c95?context=explore)
* [`1.8.1212`](https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1212/images/sha256-578202109103d1f9a4d375b49b62c229f58a7b73fb5d2c92c181786529af77a2?context=explore)
* [`1.8.1211`](https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1211/images/sha256-cf0be5fcc1710795ac632f95e897281ee0fa5ba0e2a7ad9411370e22c56d22ee?context=explore)
* [`1.8.1210`](https://hub.docker.com/layers/linuxserver/prowlarr/1.8.1210/images/sha256-a07b116ac3938306b3de1dade0a7ec2d86e754384a2eea483ae128d24ddc04f8?context=explore)

**Comment by aptalca** [comment]
_sigh_

I'm really really really tired of repeating myself over and over on this subject.

None of the tags you listed conform to the two formats I listed above as supported and promoted.
None of those have the appended `lsXX` or the prepended `version-` tags.
Therefore none of those tags are among the ones that we promote to our users as supported.

The tagging standard we set for ourselves is what is used in our ci. It regulates what tags we push for each build. It does not regulate what tags we are not supposed to push.

We already told you that we don't delete old tags unless it's absolutely necessary (and that we are able to). There are many old tags in our registries that we don't support or promote.

Your concerns are noted.

Bottom line is, we don't support or promote semver, and we don't support or recommend tools that rely on semver.

I will not repeat myself again.

Thank you.
